### PR TITLE
Add pam specific tests (2/4)

### PR DESCRIPTION
**Pull request by gitmp01**
### Relevant changes

 - Operation's amount has been changed to `asset` for better usability
 - Removed setemitter & settopiczero
 - Restored pam.test.hpp for testing
 - Revised test utils.js
 - Event id should be used as an output parameter for `check_authorization()`

### CODE REVIEW

File: cpp/contracts/adapter.cpp [hunk 1]
```diff
@@ -233,25 +200,20 @@ void adapter::settle(const name& caller, const operation& operation, const metad
...
 
    name xerc20 = search_token_bytes->xerc20;
    check(is_account(xerc20), "Not valid xerc20 name");
-   if (operation.amount > 0) {
-      auto quantity = from_wei(
-         operation.amount,
-         search_token_bytes->xerc20_symbol
-      );
+   if (operation.amount.amount > 0) {
+      asset quantity(operation.amount.amount, search_token_bytes->xerc20_symbol);
```
Comment (gitmp01) [inline cpp/contracts/adapter.cpp:216]: See `operation.hpp`

File: cpp/test/adapter-local.test.js
Comment (gitmp01) [inline cpp/test/adapter-local.test.js:1]: Sorry about this diff, but I have vscode auto-formatting enabled and is annoying

File: cpp/contracts/adapter.cpp [hunk 2]
```diff
@@ -171,57 +163,32 @@ void adapter::settee(public_key pub_key, bytes attestation) {
...
    _tee_pubkey.set(pam::tee{
-      .key = pub_key 
+      .key = pub_key
    }, get_self());
 
    // print("attestation: ")
 }
 
-void adapter::settopiczero(bytes chain_id, bytes topic_zero) {
+void adapter::setorigin(bytes chain_id, bytes emitter, bytes topic_zero) {
```
Comment (envin3) [inline cpp/contracts/adapter.cpp:172]: nice

File: cpp/contracts/pam.hpp [hunk 3]
```diff
@@ -33,7 +43,106 @@ namespace eosio {
...
+            bytes tx_id = extract_32bytes(metadata.preimage, offset);
+
+            if (block_id != operation.blockId || tx_id != operation.txId) {
+                return false;
+            }
+
+            return true;
+        }
+
+        void check_authorization(name adapter, const operation& operation, const metadata& metadata, checksum256& event_id) {
```
Comment (envin3) [inline cpp/contracts/pam.hpp:67]: this function isn't updated. We need a merge.

File: cpp/contracts/test.pam.cpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#include "test.pam.hpp"
+
+namespace eosio {
+
+void testpam::setadapter(const name& adapter) {
+   adapter_singleton _adapter(get_self(), get_self().value);
+   _adapter.set(adapter, get_self());
+}
+
+void testpam::isauthorized(const operation& operation,const metadata& metadata) {
```
Comment (envin3) [inline cpp/contracts/test.pam.cpp:10]: :+1:

**Review by envin3 — CHANGES_REQUESTED** [review]
we need a merge with https://github.com/pnetwork-association/pnetwork-v4/pull/51

**Pull request merged**